backend/gateway-service/src/modules/auth/dto/sign-up.dto.ts

```diff
@@ -0,0 +1,20 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsEmail, IsNotEmpty, IsString } from 'class-validator';
+
+export class SignUpDto {
+  @ApiProperty({
+    example: 'test@example.com',
+  })
+  @IsEmail()
+  @IsNotEmpty()
+  email: string;
+
+  @ApiProperty({
```

Comment:
In future we can discuss with the frontend to introduce more password complexities. For example:

```
@ApiProperty({
    description: 'Password must be at least 6 characters, include at least one uppercase letter, one number, and one special character',
    example: 'Password123!',
  })
  @IsString()
  @IsNotEmpty()
  @MinLength(6)
  @Matches(/(?=.*[0-9])/, { message: 'Password must contain a number' })
  @Matches(/(?=.*[A-Z])/, { message: 'Password must contain an uppercase letter' })
  @Matches(/(?=.*[!@#$%^&*])/, { message: 'Password must contain a special character' })
  password: string;
  ```

Reply:
Either way, the complexity handling should be handled on frontend